pkg/daemon/ceph/agent/agent.go

```diff
@@ -113,7 +113,9 @@ func (a *Agent) Run() error {
 		flexvolumeController,
 		volumeAttachmentController)
 	stopChan := make(chan struct{})
-	clusterController.StartWatch(v1.NamespaceAll, stopChan)
+	if err := clusterController.StartWatch(v1.NamespaceAll, stopChan); err != nil {
+		logger.Errorf("failed to start watch cluster controller . %v", err)
```

Comment:
instead of logging this error we really should return the error and fail. If the watch fails, the agent should fail.